Unity-Technologies/Graphics
# Added detail mapping in lit shader

# **Please read;**
## **PR Workflow for the Graphics repository:**
* **All PRs must be opened as draft initially**
* Reviewers can be added while the PR is still in draft
* The PR can be marked as “Ready for Review” once the reviewers have confirmed that **no more changes are needed**
* Tests will start automatically after the PR is marked as “Ready for Review”
* **Do not use [skip ci]** - this can break some of our tooling
* Read the [Graphics repository & Yamato FAQ](http://go/graphics-yamato-faq).

### Checklist for PR maker

---
### Purpose of this PR
Added detail mapping in lit shader.

---
### Testing status

**Manual Tests**: What did you do?
- Other: 

**Automated Tests**:
Added gfx test 132_DetailMapping
![image](https://user-images.githubusercontent.com/24258017/88059345-64b9e680-cb6d-11ea-89e8-f7656f88eb63.png)

**Yamato**:
https://yamato.prd.cds.internal.unity3d.com/jobs/902-Graphics/tree/universal%252Fsurface-inputs%252Fdetail-mapping

---
### Comments to reviewers

**IMPORTANT**:
Currently detail works in two possible modes:
- _DETAIL_MUL2X, works similar as builtin. Only exception that normal is blended with RNM where in builtin with WB (https://blog.selfshadow.com/publications/blending-in-detail/), where it cost only one instruction more (7v8). So it is acceptable price in my opinion for quality.
- _DETAIL_SCALED, adds albedo scaling. Compared to _DETAIL_MUL2X albedo blend it has roughly 8 more instructions.

**_DETAIL_SCALED automatically enables if albedo scaler is not 1.00. I am curious if this something sounds okey, I added in UI info box**

**UI Changes**:
![image](https://user-images.githubusercontent.com/24258017/88059555-afd3f980-cb6d-11ea-98ff-8892747197b8.png)
![image](https://user-images.githubusercontent.com/24258017/88059581-b9f5f800-cb6d-11ea-896a-77f16d420d53.png)

**For QA**:
This PR adds detail mapping into lit.shader. [Detail mapping](https://docs.unity3d.com/Manual/StandardShaderMaterialParameterDetail.html) allows adding additional color/normal details on top of the current maps.
So I think the best testing scenario would be user experience:
- Check [lit shader documentation](https://github.com/Unity-Technologies/Graphics/blob/7a060376cb7893b901f5526d40c250c9b6fc11f3/com.unity.render-pipelines.universal/Documentation~/lit-shader.md).
- Try to create your own material with lit shader and fill detail inputs.
- Investigate if everything is clear from UX side.
- Try upgrading builtin project that has standard shaders with detail inputs. Check if you get same results.

**Dependency**:
https://github.com/Unity-Technologies/Graphics/pull/1234
